### ISSUE TITLE
How do I set the initial flags when using memory-adapter?

### ISSUE THREAD

**Issue by janhoogeveen** [tdeekens/flopflip#369]
I've got memory adapter working right now with Redux. However, it looks to me like the only way to add flags to the store is by using `updateFlags({ fooFeature: true });`.

Is there a way where I can preconfigure all default flags? I want to do an API call to retrieve new flags and update those manually whenever I feel like they need changing, but that doesn't mean I want to block components from rendering while the store is not populated yet.

Relevant code:

**Redux store**

```js
import { createStore, applyMiddleware, compose } from 'redux';
import { createFlopFlipEnhancer } from '@flopflip/react-redux';
import adapter from '@flopflip/memory-adapter';
import rootReducer from './rootReducer';

const enhancer = compose(
  ...
  createFlopFlipEnhancer(adapter),
  ...
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancer);
  return store;
}
```

**Reducer setup**
```js
import { combineReducers } from 'redux';
import { flopflipReducer, FLOPFLIP_STATE_SLICE } from '@flopflip/react-redux';
import utility from './reducers/utility';

const rootReducer = combineReducers({
  [FLOPFLIP_STATE_SLICE]: flopflipReducer,
  utility,
});

export default rootReducer;
```

What I came up with on my own:

1. Using [createStore's preloadedState argument](https://redux.js.org/api-reference/createstore#arguments). This works, but it feels very wrong. This should be reserved for sending state from the server to client for state hydration. Not like a place to set default flags for my reducers.

2. Not using flopflopReducer in my combineReducer method. I don't think this will work, and you're probably using this method for a good reason.

3. Using updateFlags({}) on initial app load, before rendering the UI. This probably works, but it feels like a hack and like it would trigger new renders without any good reason.

**Comment by janhoogeveen** [comment]
I think the flopflipReducer method from `@flopflip/react-redux` could accept an _initialFlags_ argument, which would solve this whole issue?

**Comment by tdeekens** [comment]
Thanks for the issue report and sorry for the midly late response. 

Lokking at your options I‘d say that option 2. should be avoided. Using option 1. seems legit as flopflip also exports the constant of the state slice it uses as FLOPFLIP_STATE_SLICE. You should be able to use the preloadedState when creating the store. Option 3. is also valid but I would prefer 2. over it. Note also that all flags default to false if not other specified. This felt like a sane default so far. For you that means you want to optimistically render components until you notice otherwise by an API call?

Flopflip could also accept initialFlags as an argument to the reducer. With that it would become a createReducer. Under the hood both: flags and status reducer are combined where the initialFlags need to be passed forward. However it is technically the same as using preloadedState when creating the store. 

Lasly, the ConfigureFlopflip accepts defaultFlags too. This is also mentioned in the readme. The relevant code snippet is [here](https://github.com/tdeekens/flopflip/blob/master/packages/react/modules/components/configure-adapter/configure-adapter.js). 

In summary, I agree that the reducer could become an higher order reducer to pass in the initial state. I am however also not 100% sure if that is needed. Would you still see value in it? Then we can build it on a PR for you to test.

**Comment by tdeekens** [comment]
This pull request implements a higher-order reducer. Would you mind providing feedback and giving it a test in your app to see if it solves what you would hope? https://github.com/tdeekens/flopflip/pull/370

**Comment by tdeekens** [comment]
@janhoogeveen did you find the time to check the PR yet?

**Comment by tdeekens** [comment]
I must say it's interesting to "request" a feature and then not giving feedback to it's implementation. That said I tested and released it now.

**Comment by tdeekens** [comment]
>  In all honesty I was pretty much avoiding that component since the documentation makes it look like it is tightly coupled to LaunchDarkly and the memory-adapter looks decoupled from ConfigureFlopFlip.

That's very insightful and something I should tackle. Can you give me hints on where those things become unclear?

**Comment by tdeekens** [comment]
That said. What would be super duper nice is if you could spin up a PR with some, no matter how small, improvements to the docs.

**Comment by janhoogeveen** [comment]
I can do that, sure.

**Comment by tdeekens** [comment]
Glad it helped.

**Comment by janhoogeveen** [comment]
I'll make a new PR for some documentation changes, but I do think I have a question there. Let's discuss in that issue.
